### PR TITLE
Make API async so we can easily move it out of proc.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
I'm preemptively merging this in.  Let me know if there's anything you want me to change here.

File: src/Workspaces/Core/Portable/SymbolSearch/SymbolSearchOptionsProvider.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Microsoft.CodeAnalysis.SymbolSearch
+{
+    [ExportOptionProvider, Shared]
+    internal class SymbolSearchOptionsProvider : IOptionProvider
+    {
+        private readonly IEnumerable<IOption> _options = ImmutableArray.Create<IOption>(
+            SymbolSearchOptions.Enabled,
+            SymbolSearchOptions.SuggestForTypesInReferenceAssemblies,
+            SymbolSearchOptions.SuggestForTypesInNuGetPackages);
```
Comment (Pilchie) [inline src/Workspaces/Core/Portable/SymbolSearch/SymbolSearchOptionsProvider.cs]: @jasonmalinowski is it safe to have a mix of per-language options and language neutral options now?

Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/SymbolSearch/SymbolSearchOptionsProvider.cs]: Yes, that silliness is all gone now.

**Pull request merged**